sdk/core/Azure.Core/src/RequestFailedException.cs

```diff
@@ -278,10 +280,102 @@ private static void BufferResponseIfNeeded(Response response)
         }
 
         // This class needs to be internal rather than private so that it can be used by the System.Text.Json source generator
-        internal class ErrorResponse
+        internal class ErrorResponse : IJsonModel<ErrorResponse>
```

Comment:
Is this only for consistency? I just wonder if that's worth going from 2 loc to 100 since this is only used in one place internally

Reply:
Its the same lines of code in reality its just that its generated in a place you didn't see before.